File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/share/Response.java
package work.socialhub.kmastodon.entity.share;

/**
 * @author uakihir0
 */
public class Response<T> {

    private T object;

    private RateLimit limit;

    private Link link;

    public T get() {
        return object;
    }

    public void set(T object) {
        this.object = object;
    }

    public RateLimit getRateLimit() {
        return limit;
    }

    public void setRateLimit(RateLimit limit) {
        this.limit = limit;
    }

    public Link getLink() {
        return link;
    }

    public void setLink(Link link) {
        this.link = link;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/response/Response.kt
package work.socialhub.kmastodon.api.response

import work.socialhub.kmastodon.entity.share.Link
import work.socialhub.kmastodon.entity.share.RateLimit
import kotlin.js.JsExport

@JsExport
class Response<T>(
    var data: T
) {
    var limit: RateLimit? = null
    var link: Link? = null
    var json: String? = null
}
